core/client/src/main/java/alluxio/client/block/stream/PacketInStream.java

```diff
@@ -104,11 +103,11 @@ public PacketInStream(PacketReader.Factory packetReaderFactory, long id, long le
   @Override
   public int read() throws IOException {
     int bytesRead = read(mSingleByte);
-    Preconditions.checkState(bytesRead != 0);
     if (bytesRead == -1) {
       return -1;
     }
-    return (int) mSingleByte[0];
+    Preconditions.checkState(bytesRead == 1);
```

Comment:
I  thought it would be cheap (1 comparison + 1 function call + 1 branch). I added this because read(mSingleByte) can return 0 though it should not in the current implementation. 
 
If you think this can be a problem, i can do a simple local benchmark.